lib/_http_client.js

```diff
@@ -27,7 +27,8 @@ function ClientRequest(options, cb) {
   var defaultAgent = options._defaultAgent || Agent.globalAgent;
   if (agent === false) {
     agent = new defaultAgent.constructor();
-  } else if (util.isNullOrUndefined(agent) && !options.createConnection) {
+  } else if ((agent === null || agent === undefined) &&
```

Comment:
On second thought, I'll leave it to your good judgment.  Doing the explicit checks is a little more efficient than `==` because they don't generate an IC.